themes/default/content/registry/packages/eks/v3-migration/_index.md

```diff
@@ -0,0 +1,308 @@
+Its successor, the `NodeGroupV2` component is functionally equivalent and easier to operate because it does not use CloudFormation templates under the hood like `NodeGroup` did.
+
+The default node group of the `Cluster` component has been updated to use the `NodeGroupV2` component as well.
+Updates to the default node group will be done by first creating the new replacement nodes and then shutting down the old ones which will move pods to the new nodes. If you need to perform the update gracefully, please have a look at [Gracefully upgrading node groups](#gracefully-upgrading-node-groups).
+
+### Default OS of node groups updated to AL2023
+
+AWS recently deprecated the Amazon Linux 2 (AL2) operating system. It will reach end of life in June 2025, after which it will receive no more security and maintenance updates.  
+Until now, this was the OS used by node groups created with the Pulumi EKS provider.
+To align the provider with EKS best practices we’ve updated the default operating system to the new AWS-recommended option, Amazon Linux 2023 (AL2023).
+
+You can either upgrade the OS right away, or intentionally configure AL2 as the desired operating system by using the `operatingSystem` parameter that’s available on all node groups.
+
+Have a look at [Gracefully upgrading node groups](#gracefully-upgrading-node-groups) for more information around the upgrade procedure.
+
+### Gracefully upgrading node groups
+
+The `ManagedNodeGroup` component gracefully handles updates by default. EKS will:
+- boot the updated replacement nodes  
+- cordon the old nodes to ensure no new pods get launched onto them  
+- drain the old nodes one-by-one
+- shut down the empty old nodes
+
+The detailed update procedure can be seen in the [AWS docs](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-update-behavior.html).
+
+For self-managed node groups (i.e., the `NodeGroup` and `NodeGroupV2` components) you have two options:
+
+1. Update the node group in place. Pulumi does this by first creating the new replacement nodes and then shutting down the old ones which will move pods to the new nodes forcibly. This is the default behavior when node groups are updated.  
+2. Create a new node group and move your Pods to that group. Migrating to a new node group is more graceful than simply updating the node group in place. This is because the migration process taints the old node group as `NoSchedule` and drains the nodes gradually.
+
+The second option involves the following steps:
+
+* Create the replacement node group side-by-side with the existing node group. When doing this you need to make sure that the two node groups are allowed to communicate with each other. You can achieve this in the following way:
+
+
+{{< chooser language "typescript,python,go,csharp,java" >}}
```

Comment:
I had a script somewhere that updates these automatically after you edit yaml. I can share if interesting.